src/libAtomVM/opcodesswitch.h

```diff
@@ -1612,15 +1612,15 @@ static bool maybe_call_native(Context *ctx, AtomString module_name, AtomString f
             const struct GCBif *gcbif = EXPORTED_FUNCTION_TO_GCBIF(exported_bif);
             switch (arity) {
                 case 1: {
-                    *return_value = gcbif->gcbif1_ptr(ctx, 0, 0, ctx->x[0]);
+                    *return_value = gcbif->gcbif1_ptr(ctx, 0, 1, ctx->x[0]);
```

Comment:
By the way, since we have another open discussion on this topic, and the issue we experienced here turned out to be something different (did I understood this right?), I suggest removing this change from here and making this PR focused on just binary_part.